pyClient/zeth/zksnark.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+
+"""
+zk-SNARK abstraction
+"""
+
+from zeth.utils import hex_to_int
+import zeth.constants as constants
+from api.prover_pb2 import VerificationKey, ExtendedProof
+from api.util_pb2 import HexPointBaseGroup1Affine, HexPointBaseGroup2Affine
+
+import json
+from abc import (ABC, abstractmethod)
+from typing import Dict, List, Tuple, Any
+# pylint: disable=unnecessary-pass
+
+# Dictionary representing a VerificationKey from any supported snark
+GenericVerificationKey = Dict[str, Any]
+
+# Dictionary representing a Proof from any supported snark
+GenericProof = Dict[str, Any]
+
+
+class IZKSnarkProvider(ABC):
```

Comment:
Please add a line of comments to explain that this class represents the Interface that needs to be implemented by all snark providers

Reply:
Add a comment for this.